### PR TITLE
add search filter in the participants page

### PR DESCRIPTION
**Pull request by anzioka**
ezgif com-video-to-gif

search

### CODE REVIEW

**Comment by anzioka** [comment]
Screen Shot 2020-01-16 at 2 43 13 PM

File: src/containers/study/StudyParticipants.js [hunk 1]
```diff
@@ -53,15 +66,30 @@ const StudyParticipants = ({ study } :Props) => {
...
     dispatch(resetRequestState(ADD_PARTICIPANT));
     setModalOpen(true);
   };
 
+  const handleOnChange = (event :SyntheticInputEvent<HTMLInputElement>) => {
+    const { currentTarget } = event;
+    const { value } = currentTarget;
+
+    const matchingResults = participants
+      .valueSeq()
```
Comment (UnsungHero97) [inline src/containers/study/StudyParticipants.js]: I don't think you need to do `.valueSeq()`.

File: src/containers/study/StudyParticipants.js [hunk 2]
```diff
@@ -53,15 +66,30 @@ const StudyParticipants = ({ study } :Props) => {
     }
   }, [dispatch, participants, studyId]);
 
+  useEffect(() => {
+    setFilteredParticipants(participants);
+  }, [participants]);
```
Comment (UnsungHero97) [inline src/containers/study/StudyParticipants.js:67]: Would this work?

```
const participants :Map = useSelector(...);
const [filteredParticipants, setFilteredParticipants] = useState(participants);
```

**Review by UnsungHero97 — CHANGES_REQUESTED** [review]

File: src/containers/study/StudyParticipants.js (hunk 1 above)
Comment (anzioka) [reply src/containers/study/StudyParticipants.js]: You are right.

File: src/containers/study/StudyParticipants.js (hunk 2 above)
Comment (anzioka) [reply src/containers/study/StudyParticipants.js:67]: I tried that but it fails the first time the page is rendered (when `getStudyParticipants` is dispatched to fetch data). I think the reason is that `participants` is initially empty and therefore `filteredParticipants` is initialized as an empty map. After the data is fetched, `filteredParticipants` is not updated, thus the table is not rendered.

**Review by UnsungHero97 — APPROVED** [review]

**Pull request merged**